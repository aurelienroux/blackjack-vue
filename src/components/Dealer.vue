<template>
  <div class="dealer">
    <p v-if="gamePlaying">Dealer: ?</p>
    <p v-else>Dealer: {{ dealerScore }}</p>
    <div class="display">
      <div v-for="(card, index) in dealerCards" :key="index">
        <img v-if="index === 0 && gamePlaying" src="@/assets/card.png" alt="" />
        <img v-else-if="index === 0 && !gamePlaying" :src="card.image" alt="" />
        <img v-else :src="card.image" alt="dealer card" />
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Dealer',
  props: ['gamePlaying', 'dealerScore', 'dealerCards']
}
</script>

<style lang="scss">
.dealer {
  margin-bottom: 1rem;
  padding: 0 2rem;

  p {
    margin: 0.5rem 0 0.25rem;
  }
}

.display {
  display: flex;
  flex-wrap: wrap;
  min-height: 6.5rem;

  @include for-tablet-portrait-up {
    min-height: 8.5rem;
  }
}
</style>
