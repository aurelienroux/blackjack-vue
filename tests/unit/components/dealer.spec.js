import { shallowMount } from '@vue/test-utils'
import Dealer from '@/components/Dealer.vue'

function createLocalWrapper(gamePlaying = false, dealerScore = 0, dealerCards = []) {
  return shallowMount(Dealer, {
    propsData: {
      gamePlaying,
      dealerScore,
      dealerCards
    }
  })
}

const dealerCards = [
  {
    code: '5C',
    image: 'https://deckofcardsapi.com/static/img/5C.png',
    images: {
      svg: 'https://deckofcardsapi.com/static/img/5C.svg',
      png: 'https://deckofcardsapi.com/static/img/5C.png'
    },
    value: '5',
    suit: 'CLUBS'
  },
  {
    code: '3S',
    image: 'https://deckofcardsapi.com/static/img/3S.png',
    images: {
      svg: 'https://deckofcardsapi.com/static/img/3S.svg',
      png: 'https://deckofcardsapi.com/static/img/3S.png'
    },
    value: '3',
    suit: 'SPADES'
  }
]

describe('Dealer component', () => {
  describe('default state', () => {
    it('renders', () => {
      const wrapper = createLocalWrapper()
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.dealer').exists()).toBe(true)
      expect(wrapper.find('p').text()).toMatch('Dealer: 0')
    })
  })

  describe('game on', () => {
    it('renders', () => {
      const wrapper = createLocalWrapper(true, 8, dealerCards)
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('p').text()).toMatch('Dealer: ?')
      expect(wrapper.findAll('img').length).toBe(2)
      expect(
        wrapper
          .findAll('img')
          .at(0)
          .attributes('src')
      ).toBe('@/assets/card.png')
      expect(
        wrapper
          .findAll('img')
          .at(1)
          .attributes('src')
      ).toBe('https://deckofcardsapi.com/static/img/3S.png')
    })
  })

  describe('game over', () => {
    it('renders', () => {
      const wrapper = createLocalWrapper(false, 20, dealerCards)
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('p').text()).toMatch('Dealer: 20')
      expect(wrapper.findAll('img').length).toBe(2)
      expect(
        wrapper
          .findAll('img')
          .at(0)
          .attributes('src')
      ).toBe('https://deckofcardsapi.com/static/img/5C.png')
      expect(
        wrapper
          .findAll('img')
          .at(1)
          .attributes('src')
      ).toBe('https://deckofcardsapi.com/static/img/3S.png')
    })
  })
})
