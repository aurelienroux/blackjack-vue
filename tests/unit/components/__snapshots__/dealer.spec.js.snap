// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dealer component default state renders 1`] = `
<div class="dealer">
  <p>Dealer: 0</p>
  <div class="display"></div>
</div>
`;

exports[`Dealer component game on renders 1`] = `
<div class="dealer">
  <p>Dealer: ?</p>
  <div class="display">
    <div><img src="@/assets/card.png" alt=""></div>
    <div><img src="https://deckofcardsapi.com/static/img/3S.png" alt="dealer card"></div>
  </div>
</div>
`;

exports[`Dealer component game over renders 1`] = `
<div class="dealer">
  <p>Dealer: 20</p>
  <div class="display">
    <div><img src="https://deckofcardsapi.com/static/img/5C.png" alt="dealer card"></div>
    <div><img src="https://deckofcardsapi.com/static/img/3S.png" alt="dealer card"></div>
  </div>
</div>
`;
